packages/org-unit-tree/src/OrgUnitTree.component.js

```diff
@@ -8,7 +8,7 @@ import StopIcon from '@material-ui/icons/Stop';
 import ModelBase from 'd2/model/Model';
 import ModelCollection from 'd2/model/ModelCollection';
 
-import { TreeView } from '@dhis2/d2-ui-core';
+import TreeView from '@dhis2/d2-ui-core/build/es/tree-view/TreeView.component';
```

Comment:
This style of import is necessary to enable tree shaking and exclude `material-ui@0.20`?  Does it actually work or is `d2-ui-core` pulling in `material-ui` anyway since it's a dependency in `package.json`?

Reply:
When I removed all the imports in example app, except for, say, SharingDialog, then the only material-ui that was in the bundle was CircularProgress (this was when I was still importing that from d2-ui-core). So I'm pretty sure that if there is no import statement, then it doesn't get into the bundle, even though it is a dependency.

Reply:
As far as the import syntax, it may be that it treeshakes even with the import syntax like 
`import { TreeView } from '@dhis2/d2-ui-core';`

But after reading this article, I decided to be on the safe side (example with lodash-es):  https://advancedweb.hu/2017/02/07/treeshaking/

Reply:
Here is an example you can run:  https://github.com/jenniferarnesen/treeshake-test.git

The import syntax matters, big-time.

Reply:
I added this to `treeshake-test/node_modules/@dhis2/d2-ui-core/package.json`:

```
"sideEffects": [
    "*.css"
],
```

and with that change i get:

```
import Heading from '@dhis2/d2-ui-core/build/es/headings/Heading.component';
// -rw------- 1 varl varl 117K Nov  2 09:41 1.9c28a522.chunk.js

import { Heading } from '@dhis2/d2-ui-core';
// -rw------- 1 varl varl 117K Nov  2 09:42 1.7a19d5fa.chunk.js

import { Heading } from '@dhis2/d2-ui-core/build/es/index.js';
// -rw------- 1 varl varl 117K Nov  2 09:43 1.7a19d5fa.chunk.js
```

I included `@dhis2/d2-ui-core/build/es/index.js` to be absolutely sure that webpack was using the ES-bundle in the `@dhis2/d2-ui-core` case. The bundle has the same hash but different timestamps so I'm confident it's the same.

Without the `sideEffects` prop in `d2-ui-core` I get, just like the results in treeshake-test.git.

```
import Heading from '@dhis2/d2-ui-core/build/es/headings/Heading.component';
// -rw------- 1 varl varl 117K Nov  2 09:46 1.1424eb58.chunk.js

import { Heading } from '@dhis2/d2-ui-core';
// -rw------- 1 varl varl 736K Nov  2 09:47 1.120157cb.chunk.js

import { Heading } from '@dhis2/d2-ui-core/build/es/index.js';
// -rw------- 1 varl varl 736K Nov  2 09:48 1.120157cb.chunk.js
```

I think we should add the `sideEffects` prop in d2-ui components/packages and do some testing. Pretty confident it's fine in d2-ui-core, but the others is more of an open question.

Reply:
Great test cases, thanks both!  I think simplifying the import syntax will be a solid win for DX.

> but we also need to test that our code is actually side-effect free.
:+1: This is key.  If we're doing ES modules correctly it should always be side-effect free, but we might not be.

Reply:
>👍 This is key. If we're doing ES modules correctly it should always be side-effect free, but we might not be.

Agreed, although some libraries which use imports to extend runtime objects (rxjs i'm looking at you) do trigger side-effects themselves. We'll need to do some deeper testing to see the impacts when we start to claim to Webpack that we don't have side effects. 😨

Reply:
Can't wait!  😨